pkg/manager/impl/node_execution_manager.go

```diff
@@ -77,9 +76,21 @@ func (m *NodeExecutionManager) createNodeExecutionWithEvent(
 		}
 		parentTaskExecutionID = taskExecutionModel.ID
 	}
+	var parentID *uint
+	if request.Event.ParentNodeMetadata != nil {
+		parentNodeExecutionModel, err := util.GetNodeExecutionModel(ctx, m.db, &core.NodeExecutionIdentifier{
```

Comment:
do we retrieve the entire model, just to link the ID? Can we not get just the "id"

Reply:
We can. The performance improvement is tiny, as node execution record size are small. In fact you would not see any difference.